orchagent/orch.cpp

```diff
@@ -264,6 +266,28 @@ void Consumer::drain()
         m_orch->doTask(*this);
 }
 
+void ZmqConsumer::execute()
+{
+    SWSS_LOG_ENTER();
+
+    size_t update_size = 0;
+    auto table = static_cast<swss::ZmqConsumerStateTable *>(getSelectable());
+    do
+    {
+        std::deque<KeyOpFieldsValuesTuple> entries;
+        table->pops(entries);
+        update_size = addToSync(entries);
+    } while (update_size != 0);
+
+    drain();
+}
+
+void ZmqConsumer::drain()
+{
+    if (!m_toSync.empty())
+        m_orch->doTask(*this);
+}
+
```

Comment:
These two functions are very similar with the class Consumer's(execute, drain). Is it possible to reuse the same code?

Reply:
For ZmqConsumer::execute(), currently it's similer, but we have plan to use protocolbuffer for Dash related entity. after protobuffer PR merged, ZMQ will update to not send/receive KeyOpFieldsValuesTuple. so this method will update soon.

For ZmqConsumer::drain(), there are following parametric polymorphism methods, so we need the class information use corresponding method for ZmqConsumer and orginal redis based Consumer:
    virtual void doTask(Consumer &consumer) { };
    virtual void doTask(ZmqConsumer &consumer) { };

Reply:
TODO: Try improve with template.

Reply:
WIll improve after UT issue fixed.